src/js/lib/utils.ts

```diff
@@ -13,24 +13,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { pushCardState } from './routing';
+
 /**
  * Toggles opening and closing of card details
- * @param {Event} e - Click event
+ * @param {HTMLElement} target - An element within the clicked card
  */
-export function toggleCardDetails(e: Event): void {
-  e.preventDefault();
-  const target = e.target as HTMLElement;
-  const details = target.closest('.card').querySelector('.card--body') as HTMLElement;
-  const info = target
-    .closest('.card')
-    .querySelector('.card--more:not(.card--title)') as HTMLElement;
-
+export function toggleCardDetails(target: HTMLElement): void {
+  const card = target.closest('.card');
+  const details = card.querySelector('.card--body') as HTMLElement;
+  const info = card.querySelector('.card--more:not(.card--title)') as HTMLElement;
   if (info.dataset.open) {
     delete info.dataset.open;
     delete details.dataset.open;
   } else {
     info.dataset.open = 'true';
     details.dataset.open = 'true';
+    pushCardState(card.getAttribute('id'));
```

Comment:
I'd like to see two more things here: first, that when a card is clicked to open it gets scrolled into view (playing around with this I kept expecting that to happen with the URL changing and the updated loading functionality) and second that, if a card is closed and the ID is the same as the current hash, the hash is removed. The later feels important to not accidentally share a card.